### PR TITLE
Feature/mttl 521 correlation middleware

### PR DESCRIPTION
**Pull request by mikemoody-amido**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-521

## Describe this PR

Adding the Correlation middleware (plus tests) to the nuget package.

### *What changes have we introduced*

* Package target necessarily changed to be .netcore3.1 because it will host .net core 3.1 MVC specific extensions which are not accessible when targeting netstandard2.1 alone.
* Implemented test project
* Adding the Correlation middleware (plus tests) to the nuget package.
* Updated CircleCi pipeline to build and test. 

#### _Checklist_

- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: Hackney.Core.Tests/xunit.runner.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "$schema": "https://xunit.net/schema/current/xunit.runner.schema.json"
```
Comment (humulla) [inline Hackney.Core.Tests/xunit.runner.json:2]: What is this used for?

File: Hackney.Core/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Hackney.Core
+{
+    public static class Constants
```
Comment (humulla) [inline Hackney.Core/Constants.cs]: Would we not want to store the file in the middleware folder?

File: Hackney.Core.Tests/xunit.runner.json (hunk 1 above)
Comment (mikemoody-amido) [reply Hackney.Core.Tests/xunit.runner.json:2]: It's just a placeholder if you want to specify any particular config for the Xunit test runner

File: Hackney.Core/Constants.cs (hunk 2 above)
Comment (mikemoody-amido) [reply Hackney.Core/Constants.cs]: Fair point. I will move it and name it better.

**Review by amidomichael — APPROVED** [review]

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+## Link to JIRA ticket
```
Comment (evak6979) [inline pull_request_template.md:1]: Is this file necessary? :o

**Review by evak6979 — APPROVED** [review]

File: pull_request_template.md (hunk 3 above)
Comment (mikemoody-amido) [reply pull_request_template.md:1]: It's the template that pre-populates the new PR description. I added it to make it align with the other LBH repos.
